### PR TITLE
Add net training test.

### PR DESCRIPTION
**Pull request by jyegerlehner**
Pertaining to [issue 24](https://github.com/ibab/tensorflow-wavenet/issues/24), this adds a test that trains the network to learn a simple audio clip (3 tones forming a chord, at least that's what it is supposed to be). The idea is to provide broad coverage of the code and sanity check that the WaveNet is trainable. 

It runs in a couple seconds of seconds on my machine, but might not be Travis-friendly since I think it has to run on the CPU there. 

If you uncomment the print statement you can see the loss drops very close to zero, so it has successfully overfit to the 0.10 second long clip.

It would be interesting to save the generated audio (and training audio) out to .wav file to see how well it sounds.

Edit: build fails at moment because travis can't find librosa.

### CODE REVIEW

**Comment by ibab** [comment]
@jyegerlehner: There was a bug in the ci install script. If you rebase on master, the tests should now run correctly.

**Comment by ibab** [comment]
Can you replace the use of `get_arguments` with default values? (Or maybe the idea was to use the current default args?)
The problem is that this will give an error when using `nosetests` to run the tests, as the `nosetests` parameters become the ones parsed by `get_arguments`.

**Comment by jyegerlehner** [comment]
Now get_arguments() is failing. I see this failure locally if I run 
```
sh ci/test.sh
```
but succeeds if I run 
```
python ci/test_net.py
```
I'm somewhat new to python and know nothing about argparse, so will have to flounder around a bit to figure this out. Or maybe I'll just rip out the get_arguments() call and put some hard-coded params in.

**Comment by ibab** [comment]
> Or maybe I'll just rip out the get_arguments() call and put some hard-coded params in.
Sorry, that's what I meant with default values :)
argparse interferes with the testing framework, and putting hard-coded values in the test should be fine.

**Comment by jyegerlehner** [comment]
> Sorry, that's what I meant with default values :)

Ah my fault; I posted before I saw that comment (needed to refresh my browser). It was perfectly clear. Sure, that's the easiest way to go. At the expense of a bit of lost coverage: I was thinking if we use the same defaults as the rest of the software, then this test does some sanity checking on them.

**Comment by ibab** [comment]
This is a really neat test, thanks for implementing it!

**Comment by larsr** [comment]
Running the test 1200 rounds or more instead of 50 rounds does not converge. The error jumps up to 4 or five and stays there...  I've noticed similar things when training on the voice data, so perhaps there is some problem.  Nice to have it manifest itself so quickly in a test, though.

```
@@ -57,9 +57,9 @@ class TestNet(tf.test.TestCase):
         with self.test_session() as sess:
             sess.run(init)
             initial_loss = sess.run(loss)
-            for i in range(50):
+            for i in range(5000):
                 loss_val, _ = sess.run([loss, optim])
-                #print "i: %d loss: %f" % (i, loss_val)
+                if i%10==0: print("i: %d loss: %f" % (i, loss_val))
 
         # Sanity check the initial loss was larger.
         self.assertGreater(initial_loss, max_allowed_loss)

```

**Comment by ibab** [comment]
That's a nice find.
I haven't encountered this myself when training yet, but I do see it when increasing the number of steps in this test (both on the GPU and CPU).
